tapdb/supply_commit_test.go

```diff
@@ -0,0 +1,1793 @@
+}
+
+// supplyCommitTestHarness holds the necessary components for testing the
+// SupplyCommitMachine's StateMachineStore implementation.
+type supplyCommitTestHarness struct {
+	t               *testing.T
+	ctx             context.Context
+	commitMachine   *SupplyCommitMachine
+	db              sqlc.Querier
+	groupPubKey     *btcec.PublicKey
+	groupKeyBytes   []byte
+	assetSpec       asset.Specifier
+	baseGenesis     asset.Genesis
+	groupKey        *asset.GroupKey
+	batchedTreeDB   BatchedUniverseTree
+	commitTreeStore *SupplyTreeStore
+}
+
+// newSupplyCommitTestHarness creates a new test harness instance.
+func newSupplyCommitTestHarness(t *testing.T) *supplyCommitTestHarness {
+	t.Helper()
+
+	setup := setupSupplyCommitTest(t)
+	ctx := context.Background()
+
+	groupKey := &asset.GroupKey{GroupPubKey: *setup.groupPubKey}
+
+	// Use the baseGenesis ID for the initial specifier if needed, or empty
+	// ID. Let's use an empty ID for a generic harness setup.
+	spec := asset.NewSpecifierOptionalGroupPubKey(
+		asset.ID{}, setup.groupPubKey,
+	)
+
+	return &supplyCommitTestHarness{
+		t:               t,
+		ctx:             ctx,
+		commitMachine:   setup.commitMachine,
+		db:              setup.db,
+		groupPubKey:     setup.groupPubKey,
+		groupKeyBytes:   setup.groupPubKey.SerializeCompressed(),
+		assetSpec:       spec,
+		baseGenesis:     setup.baseGenesis,
+		groupKey:        groupKey,
+		batchedTreeDB:   setup.commitTreeStore.db,
+		commitTreeStore: setup.commitTreeStore,
+	}
+}
+
+// addTestMintAnchorUniCommitment inserts a mint_anchor_uni_commitments record
+// using harness data.
+func (h *supplyCommitTestHarness) addTestMintAnchorUniCommitment(batchID int64,
+	spentBy sql.NullInt64) int64 {
+
+	h.t.Helper()
+
+	internalKey := test.RandPubKey(h.t)
+	anchorCommitID, err := h.db.UpsertMintAnchorUniCommitment(
+		h.ctx, sqlc.UpsertMintAnchorUniCommitmentParams{
+			BatchID:            int32(batchID),
+			TxOutputIndex:      int32(test.RandInt[uint32]() % 100),
+			TaprootInternalKey: internalKey.SerializeCompressed(),
+			GroupKey:           h.groupKeyBytes,
+			SpentBy:            spentBy,
+		},
+	)
+	require.NoError(h.t, err)
+
+	return anchorCommitID
+}
+
+// currentState fetches the current state of the state machine via FetchState.
+func (h *supplyCommitTestHarness) currentState() supplycommit.State {
+	h.t.Helper()
+
+	state, _, err := h.commitMachine.FetchState(h.ctx, h.assetSpec)
+	require.NoError(h.t, err)
+	return state
+}
+
+// currentTransition fetches the current pending transition via FetchState.
+// Returns None if no transition is pending.
+//
+//nolint:lll
+func (h *supplyCommitTestHarness) currentTransition() lfn.Option[supplycommit.SupplyStateTransition] {
+	h.t.Helper()
+
+	_, transitionOpt, err := h.commitMachine.FetchState(h.ctx, h.assetSpec)
+	require.NoError(h.t, err)
+	return transitionOpt
+}
+
+// assertPendingTransitionExists asserts that a pending (non-finalized)
+// transition exists.
+//
+//nolint:lll
```

Comment:
We might as well add this to the top of the file so it counts for its entirety :sweat_smile:

Reply:
Heh, I prefer to just sprinkle it as needed. The linter comments don't really both me when reading/reviewing code. 

In this instance, I added it as there's no good place to split the line w/o it looking awkward. I'd rather just throw this on and move along instead of spending additional time on the more marginal aspects of our code style.